### PR TITLE
Added support for apiSecret to api_key auth mode

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe your changes
For authentication in Clari Copilot, both apiKey and apiSecret have to be passed in the request header. To achieve this, I have added a provision to enable users to add an apiSecret while creating a connection with this provider. This will, in turn, be called when authenticating the request.

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

**Comment by hassan254-prog** [comment]
clari_tweeking2

File: packages/server/lib/controllers/apiAuth.controller.ts [hunk 1]
```diff
@@ -129,10 +129,14 @@ class ApiAuthController {
             if (!req.body.apiKey) {
                 errorManager.errRes(res, 'missing_api_key');
 
+                return;
+            } else if (config.provider === 'clari-copilot' && !req.body.apiSecret) {
+                errorManager.errRes(res, 'missing_api_secret');
```
Comment (khaliqgant) [inline packages/server/lib/controllers/apiAuth.controller.ts:134]: Shouldn’t hard code a provider like this

**Comment by khaliqgant** [comment]
Closing in favor of https://github.com/NangoHQ/nango/pull/1613

**Comment by khaliqgant** [comment]
Thanks for your contribution @hassan254-prog. We didn't want to hard code the provider like that in several places but rather can use the `connectionConfig` to store the additional api key param

**Pull request closed without merging**